src/controllers/playback/videoosd.js

```diff
@@ -334,18 +334,24 @@ define(["playbackManager", "dom", "inputManager", "datetime", "itemHelper", "med
 
             if (item) {
                 var imgUrl = seriesImageUrl(item, {
+                    maxWidth: Math.round(dom.getScreenWidth() * 0.20),
```

Comment:
why `0.2`?

Reply:
Posters on the video osd are 10% width. With the 2x size to counter Skia's blurry resize, it's 20%, or 0.2 *screen width.

Reply:
I don't like that is hardcodes some internal knowledge in a way like this.
In a month or so no one would understand this magic number at all. Also a theme (or custom CSS) can change this poster width.

Can we query real poster width instead (and multiply it by DPI) instead of doing this funny workaround thing?..

Reply:
> Can we query real poster width instead (and multiply it by DPI) instead of doing this funny workaround thing?..

I think `cardbuilder` first builds urls - no card exists.

Not sure if this is possible:
* generate empty cards (with placeholders)
* set image urls

Reply:
This is indeed an oversight and fixable (at least in this case, I need to check the others).

My suggestion is:

```suggestion
                    maxWidth: Math.round(dom.getScreenWidth() * osdPoster.clientWidth),
```

**However**, this needs to be tested AND it will eventually need a polyfill for early WebOS.

Reply:
Huge poster?

Reply:
Details page build the poster before, so we can use the same technique. 

CardBuilder computes the number of cards per rows here: https://github.com/jellyfin/jellyfin-web/blob/039728359914d18765a78f22b2aaf73fc9a1aebf/src/components/cardbuilder/cardBuilder.js#L17

It then uses that to find out the size of a card, so it is already dynamic (But it also uses a ton of hardcoded values)

Reply:
More info on the support for the solution proposed: https://caniuse.com/#feat=mdn-api_element_clientwidth

Reply:
> > `dom.getScreenWidth() * osdPoster.clientWidth`
> 
> Huge poster?

Indeed, I'm dumb and I'm working on something else, so I only gave this a cursory glance :smile: 

It'd be:

```suggestion
                    maxWidth: osdPoster.clientWidth),
```